### PR TITLE
feat(CoordinateSystem): new generic coordinate system and conversion

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: bindings/python/src/mesh/helpers/crs_helper.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+namespace geode
+{
+    void define_mesh_crs_helper( pybind11::module& module )
+    {
+        PYTHON_MESH_CRS_HELPER( edged_curve, 2 );
+        PYTHON_MESH_CRS_HELPER( edged_curve, 3 );
+        PYTHON_MESH_CRS_HELPER( point_set, 2 );
+        PYTHON_MESH_CRS_HELPER( point_set, 3 );
+        PYTHON_MESH_CRS_HELPER( solid_mesh, 3 );
+        PYTHON_MESH_CRS_HELPER( surface_mesh, 2 );
```
Comment (panquez) [inline bindings/python/src/mesh/helpers/crs_helper.cpp:53]: Why not surface_mesh 3D?

File: tests/geometry/test-coordinate-system.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    check_point( coord_system, { { 0, 1 } }, { { -1, 0 } } );
+    check_point( coord_system, { { 0, 0 } }, { { -2, -1 } } );
+    check_point( coord_system, { { -1, 0 } }, { { -3, -1 } } );
+    check_point( coord_system, { { 2, 0 } }, { { 0, -1 } } );
+}
+
+void test()
+{
+    accute_system();
+    optus_system();
```
Comment (panquez) [inline tests/geometry/test-coordinate-system.cpp:72]:
```suggestion
    acute_system();
    obtuse_system();
```

File: tests/geometry/test-coordinate-system.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    OPENGEODE_EXCEPTION(
+        coordinates.inexact_equal( output ), "[Test] Wrong coordinates" );
+    const auto global_point = coord_system.global_coordinates( coordinates );
+    OPENGEODE_EXCEPTION( input.inexact_equal( global_point ),
+        "[Test] Wrong global coordinates" );
+}
+
+void accute_system()
+{
+    geode::Logger::info( "[Test] Accute system" );
```
Comment (panquez) [inline tests/geometry/test-coordinate-system.cpp:45]:
```suggestion
void acute_system()
{
    geode::Logger::info( "[Test] Acute system" );
```

File: tests/geometry/test-coordinate-system.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    };
+    check_point( coord_system, { { 0, 1 } }, { { -1, 0 } } );
+    check_point( coord_system, { { 0, 0 } }, { { 0, -1 } } );
+    check_point( coord_system, { { -1, 0 } }, { { -1, -1 } } );
+    check_point( coord_system, { { 2, 0 } }, { { 2, -1 } } );
+}
+
+void optus_system()
+{
+    geode::Logger::info( "[Test] Optus system" );
```
Comment (panquez) [inline tests/geometry/test-coordinate-system.cpp:58]:
```suggestion
void obtuse_system()
{
    geode::Logger::info( "[Test] Obtuse system" );
```

**Review by panquez — APPROVED** [review]

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**